src/polyfill/RTCPeerConnection.ts

```diff
@@ -20,110 +29,86 @@ export default class RTCPeerConnection extends EventTarget implements globalThis
-                if (config.iceServers[i] === null) throw new TypeError('IceServers cannot be null');
-                if (config.iceServers[i] === undefined) throw new TypeError('IceServers cannot be undefined');
-                if (Object.keys(config.iceServers[i]).length === 0) throw new TypeError('IceServers cannot be empty');
-
-                // If iceServers is string convert to array
-                if (typeof config.iceServers[i].urls === 'string')
-                    config.iceServers[i].urls = [config.iceServers[i].urls as string];
-
-                // urls can not be empty
-                if ((config.iceServers[i].urls as string[])?.some((url) => url == ''))
-                    throw new exceptions.SyntaxError('IceServers urls cannot be empty');
-
-                // urls should be valid URLs and match the protocols "stun:|turn:|turns:"
-                if (
-                    (config.iceServers[i].urls as string[])?.some(
-                        (url) => {
-                            try {
-                                const parsedURL = new URL(url)
-
-                                return !/^(stun:|turn:|turns:)$/.test(parsedURL.protocol)
-                            } catch (error) {
-                                return true
-                            }
-                        },
-                    )
-                )
-                    throw new exceptions.SyntaxError('IceServers urls wrong format');
-
-                // If this is a turn server check for username and credential
-                if ((config.iceServers[i].urls as string[])?.some((url) => url.startsWith('turn'))) {
-                    if (!config.iceServers[i].username)
-                        throw new exceptions.InvalidAccessError('IceServers username cannot be null');
-                    if (!config.iceServers[i].credential)
-                        throw new exceptions.InvalidAccessError('IceServers username cannot be undefined');
-                }
+    onconnectionstatechange: globalThis.RTCPeerConnection['onconnectionstatechange'];
+    ondatachannel: globalThis.RTCPeerConnection['ondatachannel'];
+    onicecandidate: globalThis.RTCPeerConnection['onicecandidate'];
+    // TODO: not implemented
+    onicecandidateerror: globalThis.RTCPeerConnection['onicecandidateerror'];
+    oniceconnectionstatechange: globalThis.RTCPeerConnection['oniceconnectionstatechange'];
+    onicegatheringstatechange: globalThis.RTCPeerConnection['onicegatheringstatechange'];
+    onnegotiationneeded: globalThis.RTCPeerConnection['onnegotiationneeded'];
+    onsignalingstatechange: globalThis.RTCPeerConnection['onsignalingstatechange'];
+    ontrack: globalThis.RTCPeerConnection['ontrack'] | null;
 
-                // length of urls can not be 0
-                if (config.iceServers[i].urls?.length === 0)
-                    throw new exceptions.SyntaxError('IceServers urls cannot be empty');
+    setConfiguration(config: RTCConfiguration): void {
```

Comment:
I for the most part removed the old implementation, because by default a lot of fallbacks are used, this is correct for W3C, but i'm not certain this is correct for NDC, this needs to be verified